joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h

```diff
@@ -382,6 +382,11 @@ update(const ros::Time& time, const ros::Duration& period)
       return;
     }
 
+    // Get state error for current joint
+    state_joint_error_.position[0] = state_error_.position[i];
```

Comment:
are you 100% sure that the indices here are correct?

Reply:
Ok I think you are right but I have a aslightly different idea for a solution.

`state_joint_error_` is truly a local variable and therefore shouldn't even be a member. The scope of it is reduced entirely to this section, it's updated by `updateStates` which is a useless update, the code could be removed from there...

let me put a commit on top of your PR here

Reply:
due to StateSegment not being particularly realtime friendly I couldn't manage a good local var fix but I think this will do